test/models/snapshot_test.rb

```diff
@@ -90,12 +90,21 @@ def test_build_snapshot_item
 
   def test_build_snapshot_item_stores_enum_database_value
     @snapshot = @snapshot_klass.first
+    post = Post.first
 
-    snapshot_item = @snapshot.build_snapshot_item(Post.first)
+    snapshot_item = @snapshot.build_snapshot_item(post)
 
     assert snapshot_item.object['status'] == 0
 
     assert_equal @snapshot.snapshot_items.first.object['status'], snapshot_item.object['status']
+
+    # Test for enum value nil
+    post.status = nil
+
+    snapshot_item = @snapshot.build_snapshot_item(post)
+
+    assert snapshot_item.object['status'].nil?
+    assert_equal @snapshot.snapshot_items.last.object['status'], snapshot_item.object['status']
```

Comment:
Why are we asserting on `@snapshot.snapshot_items`? I think we only need to assert on `snapshot_item.object['status']` in these tests?